paasta_tools/kubernetes/bin/paasta_secrets_sync.py

```diff
@@ -96,50 +111,76 @@ def main() -> None:
     secret_provider_name = system_paasta_config.get_secret_provider_name()
     vault_cluster_config = system_paasta_config.get_vault_cluster_config()
     kube_client = KubeClient()
+    services_to_k8s_namespaces = get_services_to_k8s_namespaces(
+        service_list=args.service_list, cluster=cluster, soa_dir=args.soa_dir,
+    )
+
     sys.exit(0) if sync_all_secrets(
         kube_client=kube_client,
         cluster=cluster,
-        service_list=args.service_list,
+        services_to_k8s_namespaces=services_to_k8s_namespaces,
         secret_provider_name=secret_provider_name,
         vault_cluster_config=vault_cluster_config,
         soa_dir=args.soa_dir,
-        namespace=args.namespace,
+        overwrite_namespace=args.namespace,
     ) else sys.exit(1)
 
 
+def get_services_to_k8s_namespaces(
+    service_list: List[str], cluster: str, soa_dir: str,
+) -> Mapping[str, Set]:
+    services_to_k8s_namespaces: Mapping[str, set] = defaultdict(set)
+    for service in service_list:
+        for instance_type in INSTANCE_TYPES:
+            instances = get_service_instance_list(
+                service=service,
+                instance_type=instance_type,
+                cluster=cluster,
+                soa_dir=soa_dir,
+            )
+            if instances:
+                services_to_k8s_namespaces[service].add(
+                    INSTANCE_TYPE_TO_K8S_NAMESPACE[instance_type]
+                )
+    return dict(services_to_k8s_namespaces)
```

Comment:
Why
```
    return dict(services_to_k8s_namespaces)
```
and not just
```
    return services_to_k8s_namespaces
```
?

Reply:
cast to dict since we initialise it with a `defaultdict`, and returning that might cause issues debugging